### PR TITLE
Finish lc #1, #26, #904, #977, #16, and #15

### PR DESCRIPTION
**Pull request by AlexTan331**
feat: 
1. pull latest changes from upstream
2. finish lc #1 and mark it as completed
3. finish lc #26 and mark it as completed
4. finish lc #904 and mark it as completed
5. finish lc #977 and mark it as completed
6. finish lc #16 and mark it as completed
7. finish lc #15 and mark it as completed

### CODE REVIEW

File: taojieTan/assignments/slidingwindow/lc904/lc904.py [hunk 1]
```diff
@@ -7,13 +7,54 @@
...
+  #initialize a max_fruits variable to keep track of the maximum number of fruits in the basket
+  max_fruits = 0
+
+  #loop through the array to store each character into basket
+  #if the length of basket is greater than 2 which means we just picked a third type of fruit, 
+  #we need to subtract the frequency of leftmost value (with window_start index) by one, 
+  #and in the meanwhile shifting the window_start index to right
+  #we keep subtracting until the length of basket is smaller or equal to two (use a while loop)
+  #after break out of the inner while loop, we compare larger length 
+  #after going through the entire array, we will get the final result of maximum length
```
Comment (knrodriguez) [inline taojieTan/assignments/slidingwindow/lc904/lc904.py:31]: Love the psuedo-coding. This may be a personal preference, but psuedo-code at the top tends to look cleaner. Comments in-line are more for clarifying statements about the code rather than the explanation of the code itself.

File: taojieTan/assignments/twopointers/lc1/lc1.py [hunk 2]
```diff
@@ -4,13 +4,42 @@
...
 
 
 def main():
   print(pair_with_targetsum([1, 2, 3, 4, 6], 6))
   print(pair_with_targetsum([2, 5, 9, 11], 11))
-
-
+  print(pair_with_targetsum([-12, -5, 0, 2, 9, 11], 4))
+  print(pair_with_targetsum([-12, -5, 0, 2, 9, 11], 40))
+  print(pair_with_targetsum([], 4))
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc1/lc1.py:42]: Nice!! :100:

File: taojieTan/assignments/twopointers/lc1/lc1.py [hunk 3]
```diff
@@ -4,13 +4,42 @@
...
+  while left_pointer < right_pointer:
+    temp_sum =  arr[left_pointer] + arr[right_pointer]
+    if temp_sum == target_sum:
+       return [left_pointer, right_pointer]
+    elif temp_sum > target_sum:
+      right_pointer -= 1 
+    else:
+      left_pointer += 1
+
+  return None
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc1/lc1.py:34]: Love it! The code is easier to read if the psuedo-code is above it, in my opinion.

File: taojieTan/assignments/twopointers/lc15/lc15.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+# Given an array of unsorted numbers, find all unique triplets in it that add up to zero.
+
+
+def search_triplets(arr):
+    # TODO: Write your code here
+    # input: an unsorted integer array (positive or/and negative)
+    # output: an array with unique triplets eqauls to zero
+
+    # edge cases: 1. length of array is smaller than 3 => return []
+    #            2. only zero in the array => return [0, 0, 0]
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc15/lc15.py:11]: Shows that you understand the prompt! Love it!

File: taojieTan/assignments/twopointers/lc15/lc15.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                triplets.append((arr[i], arr[left], arr[right]))
+                while left < right and arr[left] == arr[left + 1]:
+                    left += 1
+                while left < right and arr[right] == arr[right - 1]:
+                    right -= 1
+
+                left += 1
+                right -= 1
+
+    return triplets
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc15/lc15.py:55]: Nice!

File: taojieTan/assignments/twopointers/lc16/lc16.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
+# Problem Statement #
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc16/lc16.py:1]: Great work!

File: taojieTan/assignments/twopointers/lc977/lc977.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
+# Problem Statement #
```
Comment (knrodriguez) [inline taojieTan/assignments/twopointers/lc977/lc977.py:1]: Great work, you have your pseudo-code written out, functional code, and added test cases for those edge cases. :100:

**Review by knrodriguez — APPROVED** [review]
